framework/src/modules/chain/chain.js

```diff
@@ -372,7 +376,15 @@ module.exports = class Chain {
 			rewardMileStones: this.options.constants.REWARDS.MILESTONES,
 			totalAmount: this.options.constants.TOTAL_AMOUNT,
 			blockSlotWindow: this.options.constants.BLOCK_SLOT_WINDOW,
+			synchronizer: new Synchronizer(),
 		});
+
+		this.blocks.synchronizer.addStrategy(
```

Comment:
IMO Synchronizer shouldn't be aware of the mechanism registered. This means also that if the mechanisms require more dependencies, Synchronizer should need to add them to, depending entirely on its child mechanisms. I think by instantiating, adding dependencies and registering the mechanisms to the Synchronizer at bootstrap time (chain.js) makes more sense in this case. This will prevent modifying the Synchronizer in the  future if mechanisms are altered.